gui/src/JobSelector.tsx

```diff
@@ -0,0 +1,317 @@
+    } else if (status === JobStatus.CANCELED || status === JobStatus.ABANDONED) {
+        colorFill = "default";
+    }
+
+    const variant = status === JobStatus.FAILED || status === JobStatus.RUNNING ? "filled" : "outlined";
+
+    return <Chip label={label} variant={variant} color={colorFill} />;
+};
+
+const JobSelectorColumns = [
+    {
+        id: "jobId",
+        primaryLabel: "Job",
+        secondaryLabel: "ID",
+    },
+    {
+        id: "submitID",
+        primaryLabel: "Submit ID",
+    },
+    {
+        id: "submitEntity",
+        primaryLabel: "Submitted entity",
+        secondaryLabel: "ID",
+    },
+    {
+        id: "createdDt",
+        primaryLabel: "Creation date",
+    },
+    {
+        id: "status",
+        primaryLabel: "Status",
+    },
+];
+
+interface JobSelectedTableHeadProps {
+    selectAllTrue: boolean;
+    order?: "asc" | "desc";
+    orderBy?: string;
+    handleSelectAllClick?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+    handleRequestSort: (id: string, columnIndex: number) => void;
+}
+function JobSelectedTableHead({
+    selectAllTrue,
+    order,
+    orderBy,
+    handleSelectAllClick,
+    handleRequestSort,
+}: JobSelectedTableHeadProps) {
+    return (
+        <TableHead>
+            <TableRow>
+                <TableCell padding="checkbox">
+                    <Checkbox color="primary" checked={selectAllTrue} onChange={handleSelectAllClick} />
+                </TableCell>
+                {JobSelectorColumns.map((item, index) => {
+                    return (
+                        <TableCell sortDirection={orderBy === item.id ? order : false} key={index}>
+                            <TableSortLabel
+                                direction={orderBy === item.id ? order : "desc"}
+                                IconComponent={FilterList}
+                                onClick={() => handleRequestSort(item.id, index)}
+                            >
+                                {item.secondaryLabel ? (
+                                    <ListItemText primary={item.primaryLabel} secondary={item.secondaryLabel} />
+                                ) : (
+                                    item.primaryLabel
+                                )}
+                            </TableSortLabel>
+                        </TableCell>
+                    );
+                })}
+                <TableCell>Actions</TableCell>
+            </TableRow>
+        </TableHead>
+    );
+}
+
+interface JobSelectedTableRowProps {
+    row: any[];
+    isItemSelected: boolean;
+    handleCheckboxClick: (event: React.MouseEvent<unknown>, name: string) => void;
+    handleStopJob: (id: string[]) => void;
+    handleDeleteJob: (id: string[]) => void;
+}
+function JobSelectedTableRow({
+    row,
+    isItemSelected,
+    handleCheckboxClick,
+    handleStopJob,
+    handleDeleteJob,
+}: JobSelectedTableRowProps) {
+    const [id, submitID, entity, createdDt, status] = row;
```

Comment:
row is [job id, job name, entity id, entity name, submit id, creation date, status]
[string, string, string, string, string, string, number]